src/controllers/NotificationController.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Tests for Notification Controllers
+ * Send HTTP requests to Notification Controllers and check mocked responses
+ */
+
+import * as express from "express";
+import { PagoPaConfig } from "../Configuration";
+import { ControllerError } from "../enums/ControllerError";
+import { HttpErrorStatusCode } from "../enums/HttpErrorStatusCode";
+import { NotificationSubscriptionRequestType } from "../enums/NotificationSubscriptionType";
+import * as NotificationAPI from "../services/NotificationAPI";
+import { NotificationSubscriptionRequestCtrl } from "../types/controllers/NotificationSubscriptionRequestCtrl";
+import * as AppResponseConverter from "../utils/AppResponseConverter";
+import * as RestfulUtils from "../utils/RestfulUtils";
+
+export function activeNotificationsSubscription(
+  req: express.Request,
+  res: express.Response,
+  pagoPaConfig: PagoPaConfig
+): void {
+  updateNotificationsSubscription(
+    req,
+    res,
+    NotificationSubscriptionRequestType.ACTIVATION,
+    pagoPaConfig
+  ).catch();
+}
+
+export function deactiveNotificationsSubscription(
+  req: express.Request,
+  res: express.Response,
+  pagoPaConfig: PagoPaConfig
+): void {
+  updateNotificationsSubscription(
+    req,
+    res,
+    NotificationSubscriptionRequestType.DEACTIVATION,
+    pagoPaConfig
+  ).catch();
+}
+
+// Update user subscription to Notification Service
+async function updateNotificationsSubscription(
+  req: express.Request,
+  res: express.Response,
+  requestType: NotificationSubscriptionRequestType,
+  pagoPaConfig: PagoPaConfig
+): Promise<boolean> {
+  // Check input
+  const errorOrRequest = NotificationSubscriptionRequestCtrl.decode(req.params);
+  if (errorOrRequest.isLeft()) {
+    RestfulUtils.sendErrorResponse(
+      res,
+      ControllerError.ERROR_INVALID_INPUT,
+      HttpErrorStatusCode.BAD_REQUEST
+    );
+    return false;
```

Comment:
what is caller supposed to do when this function returns `false ` ? 
as a general advice, return Eithers instead of boolean as they carry the error value / cause to the caller.

```ts
Promise<Either<Error, TypeofSuccessfulResult>>
return errorOrRequest;
```

by now, it's ok code to leave the code as is.